ext/php7/coms.c

```diff
@@ -303,17 +307,14 @@ bool ddtrace_coms_threadsafe_rotate_stack(bool attempt_allocate_new, size_t min_
 }
 
 bool ddtrace_coms_buffer_data(uint32_t group_id, const char *data, size_t size) {
-    if (!data || size > DDTRACE_COMS_STACK_MAX_SIZE) {
+    // Note: payloads of size DDTRACE_COMS_STACK_MAX_SIZE are not storable as in addition to the payload we also have to
+    // store the size and the group_id: see _dd_store_data(). However, this is an acceptable approximation to avoid
+    // duplicating the same logic that we already have in _dd_store_data() or to add indirection adding a new function
+    // only to calculate the total size.
+    if (!data || size > DDTRACE_COMS_STACK_MAX_SIZE || size == 0) {
         return false;
     }
 
-    if (size == 0) {
```

Comment:
Explanation in commit https://github.com/DataDog/dd-trace-php/pull/1185/commits/98b65326de89cee6e4126efabfc7685d907a71f6